db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj

```diff
@@ -2353,9 +2353,25 @@ public class SQLParser
      * CREATE TABLE X AS [SELECT A, B, C FROM Y] WITH DATA
      */
    public String parseQueryString(String src) {
-       boolean inSection = false;
-       StringBuilder buf = new StringBuilder();
+        boolean inSection = false;
+        StringBuilder buf = new StringBuilder();
+
+        // Use a non-printable character as an end-of-hint marker.
+        char endOfHintMarker = 0x01;
+
+        // Strip out comments starting with --, except --splice-properties hints.
+        src = src.replaceFirst("(--)(?i)(?!splice-properties)(.*?)(\\n|\\r|\\r\\n)", " ");
+
+        // Add an end-of-hint marker to a --splice-properties hint, so we know where to
+        // place the newline later on.
+        src = src.replaceFirst("(--)(?i)(?=splice-properties)(.*?)(\\n|\\r|\\r\\n)",
+                               "$1$2$3"+Character.toString(endOfHintMarker));
+
+        // Strip out C-style comments:   /* This type of comment */
+        src = src.replaceFirst("(/\\*.*?\\*/)(\\n|\\r|\\r\\n)", " ");
```

Comment:
I think C-style comment does not need to end with a new line, so we don't need to put (\\n|\\r|\\r\\n) as part of the pattern.

Reply:
I think you're right... I found an issue with nested C-style comments after removing the check for newlines and carriage returns, so I may need a new regex to handle that.

Reply:
@yxia92 Java doesn't support recursive regular expressions, so I didn't find an easy way to handle nested C-style comments.  This is a low priority item, so I've opened [DB-8575](https://splicemachine.atlassian.net/browse/DB-8575) to handle this case.